### PR TITLE
Contact details ui test refactor

### PR DESCRIPTION
**Pull request by alibombaye**

### CODE REVIEW

File: app/test-utils/testFunctions/questionTests.js [hunk 1]
```diff
@@ -54,14 +54,12 @@ export const runQuestionTests = ({
...
     } else if (questionType === 'multi-question') {
-      Object.keys(sectionManifest.questions).forEach((question) => {
-        runQuestionTests({
-          pageSetup,
-          sectionManifest: sectionManifest.questions[question],
-          data,
-          sectionId,
-          sectionParent: question,
-        });
+      runQuestionTests({
```
Comment (kat1906) [inline app/test-utils/testFunctions/questionTests.js]: noice

File: app/test-utils/testFunctions/questionTests/errorTests.js [hunk 2]
```diff
@@ -11,37 +11,47 @@ const goToAnchorFromErrorSummary = ({
...
+    const questionIdBasedOnParentId = parentQuestionId ? `${parentQuestionId}[${questionIdBasedOnType}]` : questionIdBasedOnType;
+    const responseBodyBasedOnType = parentQuestionId
+      ? {
+        [parentQuestionId]: {
+          [questionIdBasedOnType]: errorType,
+        },
+      }
+      : {
+        [questionIdBasedOnType]: errorType,
+      };
```
Comment (kat1906) [inline app/test-utils/testFunctions/questionTests/errorTests.js]: this is so fun isn't it?? 😄

File: app/test-utils/testFunctions/questionTests/errorTests.js [hunk 3]
```diff
@@ -11,37 +11,47 @@ const goToAnchorFromErrorSummary = ({
...
       .put(`${apiPath}/sections/${sectionId}`)
-      .reply(400, {
-        [questionIdBasedOnType]: errorType,
-      });
+      .reply(400, responseBodyBasedOnType);
 
     const errorSummary = Selector('[data-test-id="error-summary"]');
     const errorSummaryList = Selector('.nhsuk-error-summary__list');
     const submitButton = Selector('[data-test-id="section-submit-button"]');
+    const expectedAnchorLink = `#${questionIdBasedOnParentId}`;
```
Comment (kat1906) [inline app/test-utils/testFunctions/questionTests/errorTests.js]: 👍

File: app/test-utils/testFunctions/questionTests/errorTests.js [hunk 4]
```diff
@@ -99,32 +124,57 @@ const mandatoryErrorTest = ({
...
+  parentQuestionId,
 }) => {
   if (errorType === 'required') {
     test(`should show error summary and validation for ${questionId} question indicating it is mandatory`, async (t) => {
       await pageSetup({ t });
+      const questionType = sectionManifest.questions[questionId].type;
+      const questionIdBasedOnType = questionType === 'bulletpoint-list' ? `${questionId}-1` : questionId;
+      const questionIdBasedOnParentId = parentQuestionId ? `${parentQuestionId}[${questionIdBasedOnType}]` : questionIdBasedOnType;
+      const renderedQuestionSelector = questionType === 'bulletpoint-list'
+        ? `field-error-${questionIdBasedOnParentId}`
```
Comment (kat1906) [inline app/test-utils/testFunctions/questionTests/errorTests.js]: did this work before? 😮

Comment (alibombaye) [reply app/test-utils/testFunctions/questionTests/errorTests.js]: haha I don't think it did 😆

File: app/test-utils/testFunctions/questionTests/errorTests.js (hunk 2 above)
Comment (alibombaye) [reply app/test-utils/testFunctions/questionTests/errorTests.js]: 😒 yeah amazing. Damm recursion 😡

**Pull request merged**